src/components/Form.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+const Form = ({ onAddBook }) => {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+
+  const handleAddClick = () => {
+    if (title && author && category) {
+      // Calculate the id for the new book
+      const newId = new Date().getTime();
+
+      // Create the new book object
+      const newBook = {
+        id: newId,
+        title,
+        author,
+        category,
+      };
+
+      console.log('New Book', newBook);
```

Comment:
- [ ] It will be great to remove the console.log statement. This is because it kills application performance and also displays sensitive information about your application to the console.